tests/WebApp.Tests/OpenidConfigurationTests.cs

```diff
@@ -0,0 +1,93 @@
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+
+using IdentityModel.Client;
+using IdentityServer4.Models;
+
+using Xunit;
+
+namespace Codidact.Authentication.WebApp.Tests
+{
+    public class OpenidConfigurationTests
+    {
+        private readonly HttpClient _http;
+
+        public OpenidConfigurationTests()
+        {
+            var host = new HostBuilder()
+                .ConfigureHostConfiguration(configuration =>
+                {
+                    configuration.AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        { "IdentityServer:Clients:0:ClientId", "testclient" },
+                        // Note. This is not the SHA256 value of 'testsecret' the function name is
+                        // misleading. Instead you get a string that contains the hash and some metadata.
+                        { "IdentityServer:Clients:0:ClientSecrets:0:Value", "testsecret".Sha256() },
+                        { "IdentityServer:Clients:0:AllowedGrantTypes:0", "client_credentials" },
+                        { "IdentityServer:Clients:0:AllowedScopes:0", "openid" },
+                    });
+                })
+                .ConfigureWebHost(webhost =>
+                {
+                    webhost.UseEnvironment("Testing");
+                    webhost.UseTestServer();
+                    webhost.UseStartup<Startup>();
+                })
+                .Start();
+
+            _http = host.GetTestClient();
+        }
+
+        [Fact]
+        public async Task GetOpenidConfiguration()
+        {
+            var response = await _http.GetAsync("/.well-known/openid-configuration");
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("application/json", response.Content.Headers.ContentType.MediaType);
+        }
+
+        [Fact]
+        public async Task RequestWithValidCredentials()
+        {
+            var client = new TokenClient(_http, new TokenClientOptions
```

Comment:
The `appsettings.json` file is never read because I used `new HostBuilder()` instead of `WebHost.CreateDefaultBuilder()`:

https://github.com/asynts/codidact-authentication/blob/d303a02870531a517b8313f372de004982a854ab/tests/WebApp.Tests/OpenidConfigurationTests.cs#L24

The configuration comes from an in-memory collection where client credentials are configured:

https://github.com/asynts/codidact-authentication/blob/d303a02870531a517b8313f372de004982a854ab/tests/WebApp.Tests/OpenidConfigurationTests.cs#L31